tests/test_eval_model.py

```diff
@@ -36,11 +36,79 @@ def test_output(self):
             # check totals
             self.assertEqual(score['exs'], i, "Total is incorrect")
             # accuracy should be one
-            self.assertEqual(score['accuracy'], 1, "accuracy != 1")
-            if 'rouge-1' in score:
-                self.assertEqual(score['rouge-1'], 1, 'rouge1 != 1')
-                self.assertEqual(score['rouge-2'], 1, 'rouge-2 != 1')
-                self.assertEqual(score['rouge-L'], 1, 'rouge-L != 1')
+            self.assertEqual(
+                'accuracy' in score, True, "Accuracy is missing from default"
+            )
+            self.assertEqual(score['accuracy'], 1, "Accuracy != 1")
+            self.assertEqual(
+                'rouge-1' in score, False, "Rouge is in the default metrics"
+            )
+
+    def test_metrics_all(self):
+        """Test output of running eval_model"""
+        parser = setup_args()
+        parser.set_defaults(
+            task='integration_tests',
+            model='repeat_label',
+            datatype='valid',
+            num_examples=5,
+            display_examples=False,
+            metrics='all',
+        )
+
+        opt = parser.parse_args(print_args=False)
+        str_output, valid, test = testing_utils.eval_model(opt)
+        self.assertGreater(len(str_output), 0, "Output is empty")
+
+        # decode the output
+        scores = str_output.split("\n---\n")
+
+        for i in range(1, len(scores)):
+            score = ast.literal_eval(scores[i])
+            # check totals
+            self.assertEqual(score['exs'], i, "Total is incorrect")
+            # accuracy should be one
+            self.assertEqual('accuracy' in score, True, "Accuracy is missing from all")
+            self.assertEqual(score['accuracy'], 1, "Accuracy != 1")
+            self.assertEqual('rouge-1' in score, True, "Rouge is missing from all")
+            self.assertEqual(score['rouge-1'], 1, 'rouge1 != 1')
+            self.assertEqual(score['rouge-2'], 1, 'rouge-2 != 1')
+            self.assertEqual(score['rouge-L'], 1, 'rouge-L != 1')
+
+    def test_metrics_select(self):
+        """Test output of running eval_model"""
+        parser = setup_args()
+        parser.set_defaults(
+            task='integration_tests',
+            model='repeat_label',
+            datatype='valid',
+            num_examples=5,
+            display_examples=False,
+            metrics='accuracy, rouge',
```

Comment:
O_o I don't understand how your thing handles the spaces O_o

Reply:
yeah... where are those handled? i'm also missing that